### PR TITLE
move food data into database

### PR DESCRIPTION
**Pull request by atlisg**
Asana task: [Move food data into database](https://app.asana.com/0/1192620845659633/1196695587748353/f)

### CODE REVIEW

File: components/MealForm.js [hunk 1]
```diff
@@ -80,7 +80,7 @@ const MealForm = ({ id, foodData, transportData, t }) => {
   const [distanceUnit, setDistanceUnit] = useState(distanceUnitOptions[0].value);
 
   let foodOptions = [];
-  for (let i = 0; i < foodData.length; i++) {
+  for (let i = 0; i < foodData?.length; i++) {
```
Comment (dh4u) [inline components/MealForm.js:83]: Hi Atli - what does the "?.length" change do?

File: components/MealForm.js [hunk 2]
```diff
@@ -195,8 +195,12 @@ const MealForm = ({ id, foodData, transportData, t }) => {
...
   };
 
-  const addIngredient = () => {
-    const food = foodData.find((f) => f.key === selectedIngredient.key);
+  const addIngredient = async () => {
+    setIsLoading(true);
+    const res = await fetch(`api/foods?key=${selectedIngredient.key}`);
+    const json = await res.json();
+    const food = json.foods[0];
+    console.log(food);
```
Comment (dh4u) [inline components/MealForm.js:203]: Should probably remove the console.log

**Review by dh4u — COMMENTED** [review]
Hi @atlisg  - just one question about the MealForm...

File: components/MealForm.js (hunk 1 above)
Comment (atlisg) [reply components/MealForm.js:83]: This is a nice fail safe method in the case when foodData is undefined.
It's the same as `foodData && i < foodData.length`.

**Comment by atlisg** [comment]
Decided against this for now. Might implement in the future.

**Pull request closed without merging**